CrossSectionAnalysis/cpp/CrossSectionCompute.cxx

```diff
@@ -0,0 +1,340 @@
+/*
+ * This file was created by SET [Surgeon] [Hobbyist developer].
+ */
+#include "CrossSectionCompute.h"
+#include <iostream>
+#include <thread>
+#include <mutex>
+#include <math.h> // sqrt
+#include <vector>
+
+#include <vtkMRMLModelNode.h>
+#include <vtkMRMLMarkupsCurveNode.h>
+#include <vtkMRMLSegmentationNode.h>
+#include <vtkPlane.h>
+#include <vtkCutter.h>
+#include <vtkConnectivityFilter.h>
+#include <vtkContourTriangulator.h>
+#include <vtkMatrix4x4.h> // Should not be required.
+#include <vtkMassProperties.h>
+#include <vtkMath.h> // Pi()
+
+std::mutex mtx;
+
+#define WORKER_MESSAGE(msg) mtx.lock();\
+                            std::cout << msg << std::endl;\
+                            mtx.unlock();
+
+vtkStandardNewMacro(vtkCrossSectionCompute);
+
+void vtkCrossSectionCompute::PrintSelf(ostream& os, vtkIndent indent)
+{
+    vtkObject::PrintSelf(os,indent);
+    
+    os << indent << "numberOfThreads: " << this->numberOfThreads << "\n";
+    os << indent << "inputCenterlineNode: " << this->inputCenterlineNode << "\n";
+    os << indent << "inputSurfaceNode: " << this->inputSurfaceNode << "\n";
+    os << indent << "inputSegmentID: " << this->inputSegmentID << "\n";
+}
+
+// The surface polydata is constant. Create it once only.
+bool vtkCrossSectionCompute::SetInputSurfaceNode(vtkMRMLNode * inputSurface, const std::string& inputSegmentId)
+{
+  inputSurfaceNode = inputSurface;
+  inputSegmentID = inputSegmentId;
+  if (inputSurfaceNode->IsA("vtkMRMLSegmentationNode"))
```

Comment:
If `IsA` returns true it does not guaranteed that SafeDownCast will return non-nullptr. Therefore, do not use IsA., but call SafeDownCast and check if the returned value is nullptr.